cl_dll/death.cpp

```diff
@@ -262,6 +268,18 @@ int CHudDeathNotice :: MsgFunc_DeathMsg( const char *pszName, int iSize, void *p
 	DEATHNOTICE_DISPLAY_TIME = CVAR_GET_FLOAT( "hud_deathnotice_time" );
 	rgDeathNoticeList[i].flDisplayTime = gHUD.m_flTime + DEATHNOTICE_DISPLAY_TIME;
 
+	const char *killsnd = m_pCvarKillSndPath->string;
+
+	// Play kill sound
+	if (g_PlayerInfoList[killer].thisplayer &&
+		!rgDeathNoticeList[i].iNonPlayerKill &&
+		!rgDeathNoticeList[i].iSuicide &&
```

Comment:
Are these needed given the first check?

Reply:
iSuicide check prevents the sound from playing if the player killed themselves (X killed X with grenade) or if the player died from entities (e.g. worldspawn or trigger_hurt).

iNonPlayerKill is true if victim index is -1, which appears to be impossible in HL and only used in TFC (or plugins). In any case, since victim here is non-player, the sound should still be played and the check needs to be removed.